packages/medusa-plugin-algolia/src/services/algolia.ts

```diff
@@ -1,24 +1,31 @@
-import algoliasearch from "algoliasearch"
-import { indexTypes } from "medusa-core-utils"
-import { SearchService } from "medusa-interfaces"
-import { transformProduct } from "../utils/transform-product"
+import { SearchTypes } from "@medusajs/types"
+import { SearchUtils } from "@medusajs/utils"
+import Algolia, { SearchClient } from "algoliasearch"
+import { AlgoliaPluginOptions, SearchOptions } from "../types"
+import { transformProduct } from "../utils/transformer"
 
-class AlgoliaService extends SearchService {
-  constructor(container, options) {
-    super()
+class AlgoliaService extends SearchUtils.AbstractSearchService {
+  isDefault = false
 
-    this.options_ = options
-    const { application_id, admin_api_key } = this.options_
+  protected readonly config_: AlgoliaPluginOptions
+  protected readonly client_: SearchClient
 
-    if (!application_id) {
+  constructor(_, options: AlgoliaPluginOptions) {
+    super(_, options)
+
+    this.config_ = options
+
+    const { applicationId, adminApiKey } = options
```

Comment:
suggestion: In order to keep the backward compatibility if we want, we should retrieve both the previous prop naming and the new and check both. wdyt? If you agree then it needs to be updated everywhere for all renamed props

Reply:
The search plugins will be major bumped, so I'd rather keep out the backward compatibility, because we can allow ourselves to do it.